ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -411,13 +449,16 @@ const Page: NextPage<Props> = ({
                                   </small>,
                                 ];
                               }}
-                              defaultSelected={
+                              defaultValue={
                                 source.options[opt.key]
                                   ? [source.options[opt.key]]
                                   : undefined
                               }
-                              name={`source.options.${opt.key}`}
-                              ref={register}
+                              {...register(`source.options.${opt.key}`, {
+                                onChange: (selected) => {
+                                  updateOption(opt.key, selected[0]?.key);
+                                },
+                              })}
```

Comment:
This is likely not quite right for typeaheads yet, but I can deal with typeaheads specifically as part of #2939